### PR TITLE
Move MARGIN/VMARGIN to iTermAdvancedSettingsModel

### PR DESCRIPTION
**Pull request by jcs**
A mostly mechanical change of `MARGIN` -> `[iTermAdvancedSettingsModel terminalMargin]` and `VMARGIN` -> `[iTermAdvancedSettingsModel terminalVMargin]` to allow these to be adjusted by the user (or to easily change the defaults from 5 and 2, as per #283).

It would probably be nice to move these to per-profile settings so different profiles can have different margins, but these values are used in so many places that make it hard to reach the current profile settings.

### CODE REVIEW

File: sources/iTermAdvancedSettingsModel.m [hunk 1]
```diff
@@ -134,6 +134,8 @@ + (NSString *)name { \
 DEFINE_BOOL(indicateBellsInDockBadgeLabel, YES, @"General: Indicate the number of bells rung while the app is inactive in the dock icon’s badge label");
 DEFINE_STRING(downloadsDirectory, @"", @"General: Downloads folder.\nIf set, downloaded files go to this location instead of the user’s $HOME/Downloads folder.");
 DEFINE_FLOAT(pointSizeOfTimeStamp, 10, @"General: Point size for timestamps");
+DEFINE_INT(terminalMargin, 5, @"General: Terminal Margin\nHow much space to leave between the left and right edges of the terminal.");
```
Comment (gnachman) [inline sources/iTermAdvancedSettingsModel.m]: Add this text to both terminalMargin and terminalVMargin at the end of the description, please:

You must restart iTerm2 after modifying this property.

**Comment by gnachman** [comment]
This is scary to change because it's completely metastasized. The effort to move it to a profile preference would be immense, as you note, and I think making it configurable this way is a pretty big win for a lot of people. Thanks for taking this on.

**Comment by jcs** [comment]
I've made those requested changes.

**Comment by ysupr** [comment]
Hi i really interested with this https://github.com/jaredculp/iterm2-borderless-padding/issues/19#issuecomment-273011908 

does it mean this PR we will have margin in top and left like that screenshot?

**Comment by gnachman** [comment]
@ysupr Yes, try the nightly build and see how you like it.

**Comment by JesseLeite** [comment]
@gnachman Is this still only available in the nightly build?

**Comment by luco** [comment]
This is very, very nice.

**Comment by yifanchen** [comment]
I have nightly build 3.1 - 20170303 now, in advanced setting, I changed the margin to 25 and 25, restarted the iterm, nothing changed, any ideas?

**Comment by gnachman** [comment]
@JesseLeite You can get 3.1.beta.1 from the downloads page now. The march to 3.1 is underway!

@yifanchen If you save prefs to a folder (in prefs > general), maybe it's not set to save changes back to the folder?

**Comment by JesseLeite** [comment]
@gnachman Just downloaded the beta and found the experimental setting. Love it! ❤️

**Comment by yifanchen** [comment]
@gnachman Hmm, it didn't work either. Isn't the padding/margin setting not in the place where the following image showing?

image

**Comment by JesseLeite** [comment]
@yifanchen I found it here:

**Comment by yifanchen** [comment]
@JesseLeite That works, thanks.

**Comment by jpruetting** [comment]
This is soooooo nice. Super happy this finally got added!

**Comment by aquartier** [comment]
It's look like this functionality is missing in 3.0.15 beta.

**Comment by gnachman** [comment]
It's not missing so much as it is a new feature of 3.1.

**Comment by ErraticFox** [comment]
I cannot find it anywhere and @JesseLeite photos broken...

**Comment by JesseLeite** [comment]
@ErraticFox fixed photo :)

**Comment by piotryordanov** [comment]
Is it possible to have the margins be different for every profile? or is this a global setting for item?

**Comment by gnachman** [comment]
It is a global setting. I would like to make it per-profile one day.

**Comment by KLVTZ** [comment]
For those wondering how to get rid of the title bar:

Open Preferences, click the Profiles tab, select your profile, click Window and select No Title Bar from the Style dropdown menu.

**Comment by kizerkizer** [comment]
How do you change this setting?

**Comment by gnachman** [comment]
It's easier to change it in **Prefs > Advanced > Height of top and bottom margins in terminal panes** and **Width of left and right margins in terminal panes**.

**Comment by cbartondock** [comment]
I still have a right margin in vim after setting all the margins to zero in iTerm2 preferences:
screen shot 2018-06-26 at 1 00 43 pm

Any thoughts?

**Comment by KLVTZ** [comment]
@cbartondock Maybe your .vimrc has some column configuration set?

**Comment by gnachman** [comment]
@cbartondock Your window's width is not an exact multiple of the width of a cell. If it's a fullscreen window, that is unavoidable since text size is limited to integer points size. If it's not a fullscreen window resize its width a little and it should snap to the grid and fix it.

**Comment by whistlegraph** [comment]
Was this setting removed?

**Comment by gnachman** [comment]
It is now in regular prefs, not advanced prefs.

Screen Shot 2021-04-05 at 10 08 14 PM

**Comment by mikepietruszka** [comment]
Are there any recent plans to make this per profile?

**Comment by gnachman** [comment]
> Are there any recent plans to make this per profile?

I hadn't planned to do that. What's the use case?

**Comment by mblarsen** [comment]
> > Are there any recent plans to make this per profile?
> 
> I hadn't planned to do that. What's the use case?

E.g. vim that handles its own margins. Switch to vim profile (borderless) but default profile has some margins.

**Comment by gnachman** [comment]
I'd rather not make it per-profile because I don't think it's useful enough to justify the added complexity.

**Comment by aasutossh** [comment]
image
not too sure about this but the margin still wont disappear completely, is this because of the font's dimension?

**Comment by gnachman** [comment]
There can always be a margin on the right or bottom if the window's size isn't an exact multiple of the cell size

**Comment by aasutossh** [comment]
I understand. Thank you.

**Comment by naavid-morshed** [comment]
I still have border at the right and bottom when I open emacs in iterm2. the left and bottom boder combines at bottom right and creates a small box shape. Setting both margins in appearances did not totally remove the border.

**Comment by gnachman** [comment]
@naavid-morshed box shape? Share a screen shot

**Pull request merged**